x/mongo/driver/topology/connection_test.go

```diff
@@ -993,3 +1127,24 @@ func (t *testCancellationListener) assertMethodsCalled(testingT *testing.T, numL
 	assert.Equal(testingT, numStopListening, t.numStopListening, "expected StopListening to be called %d times, got %d",
 		numListen, t.numListen)
 }
+
+// hangingTLSConn is an implementation of tlsConn that wraps the tls.Conn type and overrides the Handshake function to
+// sleep for a fixed amount of time.
+type hangingTLSConn struct {
+	*tls.Conn
+	sleepTime time.Duration
+}
+
+var _ tlsConn = (*hangingTLSConn)(nil)
```

Comment:
What is the point of making `nil` pointers like this after a `struct` definition?

Reply:
It's a compile-time assertion that `hangingTLSConn` implements the `tlsConn` interface. Not necessary, but it's good practice. This way, if a future author changes `tlsConn`, we'll immediately know that this type needs to be updated as well.

Reply:
Ah ok, thanks for the explanation. I'll start adding statements like this in the future.